### PR TITLE
Parse platform.json, hwsku.json for generate minigraph

### PR DESCRIPTION
**Pull request by dmytroxshevchuk**
### Description of PR

After transition from port_config.ini to platform.json, hwsku.json, we can`t fully deprecate and remove port_config.ini cause of port_mgmt/ansible use port_config.ini for generate minigraph. So currently we have configuration(platform.json, hwsku.json) using by sonic for configure ports and configuration(port_config.ini) using by sonic_mgmt for generate minigraph. So changes in platform.json and hwsku.json will not affect minigraph because of port_config.ini so we have configuration mismatch.
PR add functionality to sonic_mgmt for parse platform.json and hwsku.json and use this configuration for generation minigraph.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
- fully deprecate and remove port_config.ini
- avoid mismatch configurations between port_config.ini and platform.json, hwsku.json

#### How did you do it?
Add functionality for parsing platform.json, hwsku.json and use this configuration for generate minigraph.

#### How did you verify/test it?
Compare generated minigraph/config_db configurations with platform.json, hwsku.json.

### CODE REVIEW

**Comment by SavchukRomanLv** [comment]
@wangxin @yxieca could you please take a look? Thank you in advance!

**Comment by yxieca** [comment]
@jleveque @theasianpianist , the change looks good to me. Can you guys also take a look?

File: ansible/library/port_alias.py [hunk 1]
```diff
@@ -89,11 +94,98 @@ def get_portconfig_path(self, asic_id=None):
...
+                parent_intf_id = 0
+                offset = int(assigned_lane) + int(offset)
+
     def get_portmap(self, asic_id=None):
         aliases = []
         portmap = {}
         aliasmap = {}
         portspeed = {}
+
+        # if platform.json and hwsku.json are exist, than get portmap from hwsku
```
Comment (jleveque) [inline ansible/library/port_alias.py:181]: Suggest to remove the word `are`

File: ansible/library/port_alias.py [hunk 2]
```diff
@@ -89,11 +94,98 @@ def get_portconfig_path(self, asic_id=None):
...
+        platform_file = self.get_platform_path()
+        hwsku_file = self.get_hwsku_path()
+        if platform_file and hwsku_file:
+            self.get_portmap_from_platform_hwsku(platform_file, hwsku_file, aliases, portmap, aliasmap, portspeed)
+            return (aliases, portmap, aliasmap, portspeed)
+
+        # platform.json or hwsku.json is not exist so get portmap from port_config.ini
```
Comment (jleveque) [inline ansible/library/port_alias.py:188]: s/is not exist/does not exist/

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by theasianpianist — APPROVED** [review]

**Comment by theasianpianist** [comment]
Please use the testmg.sh script to verify your changes.

File: ansible/library/port_alias.py (hunk 2 above)
Comment (dmytroxshevchuk) [reply ansible/library/port_alias.py]: @jleveque done, please check

**Review by jleveque — APPROVED** [review]

**Comment by yxieca** [comment]
@dmytroxshevchuk I need to revert this PR. We found following issue with platform.json as grant truth:

For example, on a Mellanox 2700 device, if we want to deploy the 100G topology:

The platform.json's alias table:
"alias_at_lanes": "etp1a, etp1b, etp1c, etp1d",

ept1a would be used for Ethernet0's alias and that causes minigraph generation error.

With non-breakout topologies, the alias would have be etp1.

**Comment by dmytroxshevchuk** [comment]
@yxieca issue with platform.json was fixed in sonic-buildimage
issue for that: https://github.com/Azure/sonic-buildimage/issues/6024
PR for that: https://github.com/Azure/sonic-buildimage/pull/6831
So I created another PR with the same changes considering on Mellanox platform.json issue.
Please take a look:
https://github.com/Azure/sonic-mgmt/pull/3469

**Pull request merged**